redisdb/tests/test_default.py

```diff
@@ -0,0 +1,109 @@
+# (C) Datadog, Inc. 2010-2017
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+from __future__ import unicode_literals
+from distutils.version import StrictVersion
+
+from datadog_checks.redisdb import Redis
+import pytest
+import mock
+import redis
+
+from .common import PORT, PASSWORD, HOST
+
+
+# Following metrics are tagged by db
+DB_TAGGED_METRICS = [
+    'redis.persist.percent',
+    'redis.expires.percent',
+    'redis.persist',
+    'redis.keys',
+    'redis.expires',
+]
+
+STAT_METRICS = [
+    'redis.command.calls',
+    'redis.command.usec_per_call'
+]
+
+
+@pytest.mark.integration
+def test_redis_default(aggregator, redis_auth, redis_instance):
+    """
+
+    """
+    db = redis.Redis(port=PORT, db=14, password=PASSWORD, host=HOST)
+    db.flushdb()
+    db.set("key1", "value")
+    db.set("key2", "value")
+    db.setex("expirekey", "expirevalue", 1000)
+
+    redis_check = Redis('redisdb', {}, {})
+    redis_check.check(redis_instance)
+
+    # check the aggregator received some metrics
+    assert aggregator.metric_names, "No metrics returned"
+
+    # check those metrics have the right tags
+    expected = ['foo:bar', 'redis_host:{}'.format(HOST), 'redis_port:6379', 'redis_role:master']
+    expected_db = expected + ['redis_db:db14']
+    for name in aggregator.metric_names:
```

Comment:
if `metric_names` is empty (no metrics send) the test would succeed. This is what we use to do in the old tests and it caused a lot of issues. I think we should loop through a list of metrics or at least assert the number of metrics sent.

Reply:
Added `assert aggregator.metric_names` but the test is badly written :/
In this phase I tried to port the code as it was before as much as possible but definitely something to fix.